Classes/common/Indexing/CDTIndexManager.m

```diff
@@ -823,6 +758,132 @@ - (BOOL)isValidIndexName:(NSString *)indexName error:(NSError *__autoreleasing *
     return YES;
 }
 
+#pragma mark Private class methods
+
++ (FMDatabaseQueue *)databaseQueueWithDatastore:(CDTDatastore *)datastore
+                                          error:(NSError *__autoreleasing *)error
+{
+    NSString *dir = [datastore extensionDataFolder:kCDTExtensionName];
+    [[NSFileManager defaultManager] createDirectoryAtPath:dir
+                              withIntermediateDirectories:TRUE
+                                               attributes:nil
+                                                    error:nil];
+    NSString *filename = [NSString pathWithComponents:@[ dir, @"indexes.sqlite" ]];
+
+    FMDatabaseQueue *database = nil;
+    NSError *thisError = nil;
+    BOOL success = YES;
+
+    if (success) {
+        database = [[FMDatabaseQueue alloc] initWithPath:filename];
+
+        success = (database != nil);
+        if (!success) {
+            NSDictionary *userInfo = @{
+                NSLocalizedDescriptionKey :
+                    NSLocalizedString(@"Problem opening or creating database.", nil)
+            };
+            thisError = [NSError errorWithDomain:CDTIndexManagerErrorDomain
+                                            code:CDTIndexErrorSqlError
+                                        userInfo:userInfo];
+        }
+    }
+
+    if (success) {
+        id<CDTEncryptionKey> encryptionKey = [datastore copyEncryptionKey];
+
+        success = [CDTIndexManager configureDatabase:database withEncryptionKey:encryptionKey];
+
+        if (!success) {
+            NSDictionary *userInfo = @{
+                NSLocalizedDescriptionKey :
+                    NSLocalizedString(@"Problem configuring database with encryption key", nil)
+            };
+            thisError = [NSError errorWithDomain:CDTIndexManagerErrorDomain
+                                            code:CDTIndexErrorEncryptionKeyError
+                                        userInfo:userInfo];
+        }
+    }
+
+    if (success) {
+        success = [CDTIndexManager updateSchema:VERSION inDatabase:database];
+
+        if (!success) {
+            NSDictionary *userInfo = @{
+                NSLocalizedDescriptionKey :
+                    NSLocalizedString(@"Problem updating database schema.", nil)
+            };
+            thisError = [NSError errorWithDomain:CDTIndexManagerErrorDomain
+                                            code:CDTIndexErrorSqlError
+                                        userInfo:userInfo];
+        }
+    }
+
+    if (!success) {
+        database = nil;
+
+        if (error) {
+            *error = thisError;
+        }
+    }
+
+    return database;
+}
+
++ (BOOL)configureDatabase:(FMDatabaseQueue *)database
+        withEncryptionKey:(id<CDTEncryptionKey>)encryptionKey
+{
+    __block BOOL success = YES;
+
+    NSString *key = [encryptionKey encryptionKeyOrNil];
+    if (key) {
+        [database inDatabase:^(FMDatabase *db) {
+          success = [db setKey:key];
```

Comment:
'setKey:' is always available, we don't need to check if the instance responds to this message. However, if we call this method without SQLCipher, it will return NO.